projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -18,3 +20,180 @@ function jetpack_blogging_prompts_add_meta_data( $keys ) {
 	$keys[] = '_jetpack_blogging_prompt_key';
 	return $keys;
 }
+
+add_filter( 'rest_api_allowed_public_metadata', 'jetpack_blogging_prompts_add_meta_data' );
+
+/**
+ * Sets up a new post as an answer to a blogging prompt.
+ *
+ * Called on `wp_insert_post` hook.
+ *
+ * @param int $post_id ID of post being inserted.
+ * @return void
+ */
+function jetpack_setup_blogging_prompt_response( $post_id ) {
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	$prompt_id = isset( $_GET['answer_prompt'] ) && absint( $_GET['answer_prompt'] ) ? absint( $_GET['answer_prompt'] ) : false;
+
+	if ( ! jetpack_is_new_post_screen() || ! $prompt_id ) {
+		return;
+	}
+
+	if ( jetpack_is_valid_blogging_prompt( $prompt_id ) ) {
```

Comment:
Ideally we could fetch the prompt by id based on the value of `answer_prompt`, but the current API endpoint doesn't work that way. So for now, I'm just checking to make sure the prompt id is known, before recording an answer.

You bring up a good point about outdated links--I can have the API request look back one day to increase the possibility of finding the prompt id.

Reply:
> but the current API endpoint doesn't work that way

Ok, makes sense

> I can have the API request look back one day to increase the possibility of finding the prompt id

Probably doesn't make sense to add the complexity though, I also think that would be a rare edge case

I think the ideal solution would be what you suggested fetching the prompt based on the ID from answer_prompt but if that doesn't exist yet, I don't think it's critical to add that. Possibly it could be a future improvement, I think it would make sense for the API to have that method.